#18 base building

jetenergy opened the pull request.

jetenergy committed:
Added the beginning of base building

jetenergy committed:
converted IBaseRoom to BaseRoom abstract class
Added lots of Tiles including a debug for the sphere collider
Changed UI buttons

jetenergy committed:
Added rooms, moved some stuff around, added debugging to show what tile it is when ingame

jetenergy committed:
Added rooms, moved some stuff around, added debugging to show what tile it is when ingame

jetenergy committed:
Added Intermediate - UnitRoom Room.
Updated some prefabs, Added the addition and removing of tiles from "rooms"
Added updating "merging" rooms

jetenergy committed:
Moved prefabs to Resources for access within code without the use of the unity UI.
added spawning of units
Fixed mouse click detection
Added cutting away navmesh when wall
Added temporary ant for testing with navmesh and walls

jetenergy committed:
Added a teleporter

jetenergy committed:
Added removing a unit from a group, removing max units from a group when room destroyed
Added a temporary death plane

jetenergy committed:
Fixed scene

jetenergy committed:
Added a bunch of null checks, renamed a couple of variables
Added Astar for splitting rooms

jetenergy committed:
converted astar to use baseUnitRooms instead for cleaner BaseTiles

jetenergy committed:
Merge branch 'Develop' into #18-BaseBuilding

# Conflicts:
#	Hivemind/Assets/Prefabs/UI/IngameUI.prefab
#	Hivemind/Assets/Scripts/TestScript.cs
#	Hivemind/Assets/Scripts/Ui/UiController.cs
#	Hivemind/Assets/Scripts/UnitController.cs
#	Hivemind/Assets/Scripts/UnitGroupList.cs

jetenergy committed:
Moved a hole load of stuff, changed some prefabs, fixed some post merge conflicts

jetenergy committed:
Cleaned up base building and removed most test objects
Fixed ant to change depending on the room changes

jetenergy committed:
Moved everything

jetenergy committed:
Merge branch 'Develop' into #18-BaseBuilding

# Conflicts:
#	Hivemind/Assets/Resources/Prefabs/QueenAnt.prefab

jetenergy committed:
Fixed "Map" specific error error

ErikLouwerse commented on Hivemind/Assets/Scripts/BaseBuilding/BaseTile.cs:
```diff
@@ -0,0 +1,93 @@
+        RoomScript = CurrTile.GetComponent<BaseRoom>();
+    }
+
+    internal void DestroyRoom()
```
Recommend deleting this method and defaulting bool forced to false in the method below

ErikLouwerse commented on Hivemind/Assets/Scripts/BaseBuilding/Rooms/BaseRoom.cs:
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Remove unused usings pls

ErikLouwerse commented on Hivemind/Assets/Scripts/BaseBuilding/Rooms/BaseUnitRoom.cs:
```diff
@@ -0,0 +1,188 @@
+                GameObject container = GameObject.Find("Ants");
+                if (container == null)
+                {
+                    throw new Exception("The \"Ants\" container was not found. Please add a container to save the Hiearchy from clutter!");
```
Hierarchy not Hiearchy

ErikLouwerse commented on Hivemind/Assets/Scripts/BaseBuilding/UnitRoomPriorityQueue.cs:
```diff
@@ -0,0 +1,29 @@
+
+class UnitRoomPriorityQueue
+{
+    List<BaseUnitRoom> queue = new List<BaseUnitRoom>();
```
queue should be private

ErikLouwerse commented on Hivemind/Assets/Scripts/Ui/UiController.cs:
```diff
@@ -20,15 +20,13 @@ public class UiController : MonoBehaviour, IInitializePotentialDragHandler, IDra
 
     [SerializeField] private TextMeshProUGUI resourceTextBox;
 
-    [SerializeField] private UnitController unitController;
```
Remove autors comment block above pls (can't comment there, not relevant to this line)

ErikLouwerse replied:
Same goes for al other author comment in the solution

ErikLouwerse commented on Hivemind/Assets/Scripts/UnitController.cs:
```diff
@@ -42,8 +45,83 @@ public void SetMaxUnits(Guid unitGroupId, int amount)
+                throw new Exception("The ant doesnt have the required script!");
+            }
+        }
+
```
Empty line

ErikLouwerse commented on Hivemind/Assets/Scripts/UnitGroup.cs:
```diff
@@ -29,28 +29,59 @@ internal UnitGroup(GameObject unitIconBase)
     }
 
-    public void SetMaxUnits(int amount)
+    public bool RemoveMax()
```
Change to void pls, same goes for methods below if they are not used as bools

ErikLouwerse commented on Hivemind/Assets/Scripts/UnitGroup.cs:
```diff
@@ -29,28 +29,59 @@ internal UnitGroup(GameObject unitIconBase)
+
+    internal void MergeGroupIntoThis(UnitGroup other)
+    {
+        this.MaxUnits += other.MaxUnits;
```
this. is not useful pls remove

ErikLouwerse commented on Hivemind/Assets/Scripts/UnitGroupList.cs:
```diff
@@ -83,15 +104,23 @@ internal void MoveUnit(UnitGroup unit, GameObject groupGameObject)
+
+        UnityEngine.Object.Destroy(unitGroup.Ui_IconObj);
+    }
+ public MindGroup GetMindGroupFromIndex(int Index)
```
spacing

MMCQ33 reviewed (changes requested):
fix erik comments

jetenergy committed:
Changed Ui buttons
Fixed comments
Added base trigger
Fixed able to place room next to queen

jetenergy committed:
ongoing Testing with Worker ant

jetenergy committed:
Added base minimap
Fixed gathering: movement jitter, leaving base, calculating resource distance from exit when in base

jetenergy committed:
Changed default scouting = false
